observability/opentelemetry-api/src/main/java/org/wildfly/extension/opentelemetry/api/WildFlyOpenTelemetryConfig.java

```diff
@@ -19,38 +20,56 @@ public final class WildFlyOpenTelemetryConfig implements OpenTelemetryConfig {
+    public static final String OTEL_EXPORTER_OTLP_TRACES_PROTOCOL = "otel.exporter.otlp.traces.protocol";
+    public static final String OTEL_LOGS_EXPORTER = "otel.logs.exporter";
     public static final String OTEL_METRICS_EXPORTER = "otel.metrics.exporter";
     public static final String OTEL_SDK_DISABLED = "otel.sdk.disabled";
     public static final String OTEL_SERVICE_NAME = "otel.service.name";
+    public static final String OTEL_SPAN_PROCESSOR_TYPE = "wildfly.span.processor.type";
     public static final String OTEL_TRACES_EXPORTER = "otel.traces.exporter";
     public static final String OTEL_TRACES_SAMPLER = "otel.traces.sampler";
     public static final String OTEL_TRACES_SAMPLER_ARG = "otel.traces.sampler.arg";
+
     private final Map<String, String> properties;
 
+    public WildFlyOpenTelemetryConfig(Map<String, String> properties) {
+        this.properties = Collections.unmodifiableMap(properties);
+    }
+
     public WildFlyOpenTelemetryConfig(String serviceName, String exporter, String endpoint,
                                       Long batchDelay, Long maxQueueSize, Long maxExportBatchSize,
-                                      Long exportTimeout, String sampler, Double ratio) {
+                                      Long exportTimeout, String spanProcessorType, String sampler, Double ratio) {
         Map<String, String> config = new HashMap<>();
         // Default to on
         addValue(config, OTEL_SDK_DISABLED, "false");
         addValue(config, OTEL_EXPERIMENTAL_SDK_ENABLED, "true");
 
         addValue(config, OTEL_SERVICE_NAME, serviceName);
-        addValue(config, OTEL_METRICS_EXPORTER, "none");
         addValue(config, OTEL_TRACES_EXPORTER, exporter);
-        switch (exporter) {
-            case "jaeger": addValue(config, OTEL_EXPORTER_JAEGER_ENDPOINT, endpoint);
-                           addValue(config, OTEL_EXPORTER_JAEGER_TIMEOUT, exportTimeout);
-                           break;
-            case "otlp": addValue(config, OTEL_EXPORTER_OTLP_ENDPOINT, endpoint);
-                         addValue(config, OTEL_EXPORTER_OTLP_TIMEOUT, exportTimeout);
-                         break;
+
+        // We don't support otel logging or metrics, so we have to set these to "none" to prevent the autoconfiguration
+        // API from trying to configure them.
+        addValue(config, OTEL_LOGS_EXPORTER, "none");
+        addValue(config, OTEL_METRICS_EXPORTER, "none");
+
+        if (exporter.equals("otlp")) {
+            addValue(config, OTEL_EXPORTER_OTLP_ENDPOINT, endpoint);
+            addValue(config, OTEL_EXPORTER_OTLP_PROTOCOL, "grpc");
+            addValue(config, OTEL_EXPORTER_OTLP_TIMEOUT, exportTimeout);
+        } else {
+            Logger.getLogger(WildFlyOpenTelemetryConfig.class)
+                    .warn("An unexpected exporter type was found: " +
+                            exporter + ". Using the default.");
```

Comment:
This should fail with an IAE. The Stage.MODEL code should correctly handle parameter validation and this class should reject any mistakes it makes.

Or, you could remove the exporter param altogether and hard-code the otlp behavior.

Reply:
Done